frontend/src/js/controllers/changePwdCtrl.js

```diff
@@ -7,67 +7,73 @@
         // function to change password
-        vm.changePassword = function() {
+        vm.changePassword = function(resetconfirmFormValid) {
+          if(resetconfirmFormValid){
             vm.loginContainer = angular.element('.change-passowrd-card');
-
+            vm.startLoader("Changing Your Password");
             var parameters = {};
             parameters.url = 'auth/password/change/';
             parameters.method = 'POST';
             parameters.data = {
                 "old_password": vm.user.old_password,
                 "new_password1": vm.user.new_password1,
                 "new_password2": vm.user.new_password2,
+                "uid": $state.params.user_id,
+                "token": $state.params.reset_token,
             }
-            parameters.token = userKey;
-            var passwordCheck = validationSvc.password_check(parameters.data.new_password1, parameters.data.new_password2);
-            if (passwordCheck.status) {
                 parameters.callback = {
                     onSuccess: function(response) {
                         var status = response.status;
                         var response = response.data;
                         vm.user.error = false;
                         console.log("PASSWORD CHANGED SUCCESSFULLY");
                         console.log(response);
+                        vm.stopLoader();
                         // navigate to challenge page
                         // $state.go('web.challenge-page.overview');
                     },
                     onError: function(response) {
                         var status = response.status;
                         var error = response.data;
                         vm.user.error = "Failed";
-                        if (status == 400) {
-                            console.log("ERROR Occured");
-                            console.log(error);
-                            angular.forEach(error, function(value, key) {
-                                if (key == 'old_password') {
-                                    vm.isValid.old_password = true;
-                                    vm.wrnMsg.old_password = value[0];
-                                }
-                                if (key == 'new_password1') {
-                                    vm.isValid.new_password1 = true;
-                                    vm.wrnMsg.new_password1 = value[0];
-                                }
-                                if (key == 'new_password2' || key == 'non_field_errors') {
-                                    vm.isValid.new_password2 = true;
-                                    vm.wrnMsg.new_password2 = value[0];
-                                }
-                            })
+                        var token_valid, oldpassword_valid ,password1_valid, password2_valid;
+                        try {
+                            token_valid = typeof(response.data.token) !== 'undefined' ? true : false;
```

Comment:
No need to check token validation error since this change password view can be open after successful login. 
Remove all `token_valid`.